### DISCUSSION TITLE
マークダウンからディレクトリ構造をツリー形式で生成してくれるもの

### DISCUSSION THREAD

**Discussion by pss-aileen** [first-contributions-ja/first-contributions-ja.github.io#65] [90: 知識・知見の共有]
理想系

```markdown
src/
├─app
|   ├─page.tsx
|   ├─layout.tsx
|   ├─loading.tsx
|   └─not-found.tsx
|
├─components
|   ├─header.tsx
|   ├─footer.tsx
|   └─ui
|     └─button.tsx
```

- https://tree.nathanfriend.io/
- https://www.text-tree-generator.com/

**Comment by pss-aileen** [comment]
@kazzyfrog さん、↑のディレクトリ構造のツリー形式のテキスト？ってどのように作られましたか？
テキストから生成できるようなgenerator探しているのですが
コンパクトでないものが多くて↓
このような感じで `-` が長いというか...
もし簡単にうまいこと @kazzyfrog さんの作られたテキストみたいに生成できる何かがあれば教えていただきたいです！

```
Edit me to generate
├── a
│   └── nice
│       └── tree
│           ├── diagram!
│           └── :)
└── Use indentation
    ├── to indicate
    │   ├── file
    │   ├── and
    │   ├── folder
    │   └── nesting.
    └── You can even
        └── use
            ├── markdown
            └── bullets!
```

```
/
├── bin
├── boot
│   └── grub
│       ├── fonts
│       └── locale
├── cdrom
└── dev
    ├── block
    ├── bsg
    └── bus
        ├── usb
        ├── 001
        └── 002
```

**Comment by pss-aileen** [comment]
あ〜それか自分で作ればいいのか...

**Comment by kazzyfrog** [comment]
自分は、よく使いまわしているのがあったので、それをベースに手書きで作成しました！

確かに、「テキストから生成できるようなgenerator」あったら良さそうですよね！

**Reply by pss-aileen** [reply]
ありがとうございます！

なるほど、そうなんですね！
確かに、普通の使い回しでも良いですね...！

ジェネレーター、こちら個人で作るか、他に作りたい方がいらっしゃれば作ったり色々ありかな〜と思いました。

いつか個人で作るかもです...！
自分が自分用に欲しい...笑

**Reply by kazzyfrog** [reply]
面白そうですね！

「ディレクトリ構成図」で調べてみたら、結構いろんな方法で生成できるっぽいです：
- [macのtreeコマンド](https://zenn.dev/chamo/articles/64d5ef76df555a)で生成
- [VSCode](https://qiita.com/Ryo-0131/items/71452423f9698b071806)の拡張で制する方法

色々試しつつ、ジャストフィットするのがなかったら作るのもありですね！
chatGPT使ったり、VSCodeの拡張にしたりも出来そうです！
